File: FrozenApp/src/main/java/com/github/lakeshire/frozen/SubFragment.java
package com.github.lakeshire.frozen;


import android.view.View;
import android.widget.RelativeLayout;

import java.util.ArrayList;
import java.util.List;

import in.srain.cube.views.ptr.PtrFrameLayout;
import lakeshire.github.com.frozenframework.fragment.BasePullFragment;
import lakeshire.github.com.frozenframework.util.ScreenUtil;
import lakeshire.github.com.frozenframework.view.focusimage.Banner;
import lakeshire.github.com.frozenframework.view.focusimage.FocusImageAdapter;
import lakeshire.github.com.frozenframework.view.focusimage.FocusImageView;

public class SubFragment extends BasePullFragment {
    private FocusImageAdapter mFocusImageAdapter;
    private FocusImageView mLayoutFocus;
    private List<Banner> mFocusImages = new ArrayList<>();


    @Override
    public int getContainerLayoutId() {
        return R.layout.fra_sub;
    }

    @Override
    public void loadData() {
        super.loadData();
    }

    @Override
    public void initUi() {
        super.initUi();

        mLayoutFocus = findView(R.id.banner);
        int screenWidth = ScreenUtil.getScreenWidth(getActivity());
        int height = (int) (screenWidth / 2.5f);
        RelativeLayout.LayoutParams lp = new RelativeLayout.LayoutParams(screenWidth, height);
        mLayoutFocus.setLayoutParams(lp);

        mFocusImages.clear();
        Banner banner1 = new Banner();
        banner1.res = R.drawable.banner_1;
        Banner banner2 = new Banner();
        banner2.res = R.drawable.banner_2;
        mFocusImages.add(banner1);
        mFocusImages.add(banner2);

        mFocusImageAdapter = new FocusImageAdapter(getActivity().getApplicationContext(),
                mFocusImages, screenWidth);
        mLayoutFocus.setAdapter(mFocusImageAdapter);
    }

    @Override
    protected void onRefresh(PtrFrameLayout frame) {
        refreshComplete();
    }

    @Override
    protected boolean checkCanRefresh(PtrFrameLayout frame, View content, View header) {
        return mLayoutFocus.canOutsideRefresh();
    }

    @Override
    protected String getTitle() {
        return "子页面";
    }

    @Override
    protected int getActionRes() {
        return 0;
    }

    @Override
    protected View.OnClickListener getActionListener() {
        return null;
    }
}

File: FrozenApp/src/main/java/com/github/lakeshire/frozen/SubFragment.kt
package com.github.lakeshire.frozen


import `in`.srain.cube.views.ptr.PtrFrameLayout
import android.content.DialogInterface
import android.view.View
import android.widget.RelativeLayout
import lakeshire.github.com.frozenframework.fragment.BasePullFragment
import lakeshire.github.com.frozenframework.util.DialogUtil
import lakeshire.github.com.frozenframework.util.ScreenUtil
import lakeshire.github.com.frozenframework.view.focusimage.Banner
import lakeshire.github.com.frozenframework.view.focusimage.FocusImageAdapter
import lakeshire.github.com.frozenframework.view.focusimage.FocusImageView
import java.util.*

class SubFragment : BasePullFragment() {
    private var mFocusImageAdapter: FocusImageAdapter? = null
    private var mLayoutFocus: FocusImageView? = null
    private val mFocusImages = ArrayList<Banner>()

    override val containerLayoutId: Int
        get() = R.layout.fra_sub

    override fun loadData() {

    }

    override fun initUi() {
        super.initUi()

        mLayoutFocus = mContainerView.findViewById(R.id.banner)
        val screenWidth = ScreenUtil.getScreenWidth(activity)
        val height = (screenWidth / 2.5f).toInt()
        val lp = RelativeLayout.LayoutParams(screenWidth, height)
        mLayoutFocus!!.layoutParams = lp

        mFocusImages.clear()
        val banner1 = Banner()
        banner1.res = R.drawable.banner_1
        val banner2 = Banner()
        banner2.res = R.drawable.banner_2
        mFocusImages.add(banner1)
        mFocusImages.add(banner2)

        mFocusImageAdapter = FocusImageAdapter(activity.applicationContext,
                mFocusImages)
        mLayoutFocus!!.adapter = mFocusImageAdapter
    }

    override fun refresh() {

    }

    override fun onRefresh() {
        refreshComplete()
    }

    override fun checkCanRefresh(frame: PtrFrameLayout, content: View, header: View): Boolean {
        return mLayoutFocus!!.canOutsideRefresh()
    }

    override fun getTitle(): String {
        return "子页面"
    }

    override fun getActionRes(): Int {
        return 0
    }

    override fun getActionListener(): View.OnClickListener? {
        return null
    }

    override fun onBackPressed(): Boolean {
        DialogUtil.showTips(activity, "确定退出？", "确定退出？", "确定", DialogInterface.OnClickListener { dialog, which -> endFragment() }, "取消", null)
        return true
    }
}
